qiskit/circuit/library/generalized_gates/uc_pauli_rot.py

```diff
@@ -13,39 +13,36 @@
 # The structure of the code is based on Emanuel Malvetti's semester thesis at ETH in 2018,
 # which was supervised by Raban Iten and Prof. Renato Renner.
 
-"""
-(Abstract) base class for uniformly controlled (also called multiplexed) single-qubit rotations R_t.
-This class provides a basis for the decomposition of uniformly controlled R_x,R_y and R_z gates
-(i.e., for t=x,y,z). These gates can have several control qubits and a single target qubit.
-If the k control qubits are in the state ket(i) (in the computational bases),
-a single-qubit rotation R_t(a_i) is applied to the target qubit for a (real) angle a_i.
-"""
+"""Uniformly controlled Pauli rotations."""
+
+from __future__ import annotations
```

Comment:
This is more ignorance on my part. Why is this needed?

Reply:
This is needed to enable type hints like `list[...]` (instead of `typing.List[...]`) or items from `collections.abc` instead of using the `typing` module in Python 3.8. For example the following code fails in Python 3.8
```python
# in 3.8 this import is needed, in 3.9 it is implied and can be removed
# from __future__ import annotations

def f(x: list[int]) -> None:
    print(x)

f([1, 2])
```